cmd/rocsp-tool/client_test.go

```diff
@@ -79,6 +83,10 @@ func TestGetStartingID(t *testing.T) {
 }
 
 func TestStoreResponse(t *testing.T) {
+	// TODO(#6517) remove this block.
+	if os.Getenv("BOULDER_CONFIG_DIR") == "test/config" {
```

Comment:
nit: I personally prefer `if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {`. It's a slightly clearer (to me!) way to say "this test only works in config-next mode".

But more importantly, I don't think that duplicating the tests in this fashion is truly necessary. It appears that their bodies are identical, except for the calls to `makeClient()` vs `makeClusterClient()`. I think it would be reasonable -- and both easier to review and easier to prove correct -- for these test functions to instead do:
```go
	if os.Getenv("BOULDER_CONFIG_DIR") == "test/config" {
		redisClient, clk = makeClusterClient()
	} else {
        redisClient, clk = makeClient()
    }
```

I know we talked about the "duplicate things" vs "do it inline" tradeoffs during standup, and said that the "duplicate stuff" approach was probably better in this case. During that discussion, I was imagining that the diffs within each codepath would be much larger than they actually are, thanks to this very-elegant PR. So I lean the other direction for these tests -- I still think you did exactly the right thing for rocsp.go and rocspcluster.go.